app/models.py

```diff
@@ -2064,6 +2064,16 @@ def set_approval(self, approved):
 
             db.session.flush()
 
+            # associate supplier with digital marketplace framework
+            framework = Framework.query.filter(
+                Framework.slug == 'digital-marketplace'
+            ).first()
+            SupplierFramework(
+                supplier_code=supplier.code,
+                framework_id=framework.id,
+                declaration={}
+            )
```

Comment:
This SupplierFramework needs to be `add`ed to the session doesn't it?

`db.session.add(sf)`

Maybe add a test to confirm it's there?